### ISSUE TITLE
Feature: HttpTrigger POCO with constructor parameters

### ISSUE THREAD

**Issue by CasperWSchmidt** [Azure/azure-webjobs-sdk-extensions#627]
AFAIK Newtonsoft.JSON is used for deserializing http requests when using a POCO as a parameter instead of HttpRequest. NewtonSoft.JSON supports contructors with parameters and I believe that Azure Functions should do to. Request objects should IMHO be read-only after the framework has created it, but this is not the case with the current requirement of parameterless constructors and public property-setters.

**Comment by v-anvari** [comment]
Hi @CasperWSchmidt, Would like to know if you are looking out for something similar to [#584](https://github.com/Azure/azure-webjobs-sdk-extensions/issues/584).

**Comment by CasperWSchmidt** [comment]
@v-anvari not exactly. I don't need both a POCO and HttpRequest object. What I'm looking for is the possibility of having the POCO being immutable (all properties being get-only) after initialization (like the new record type in C#). When I submitted this issue constructor parameters were not supported. I'm not sure if it is now?

**Comment by v-anvari** [comment]
Hi @CasperWSchmidt, Could you give us a sample code or example that you are trying to figure out. 

Kindly give us more insights on what you are looking out for.

**Comment by CasperWSchmidt** [comment]
@v-anvari I'm looking for the possiblity of using the following POCO:
```
public class SomeCommand
{
    public string SomeProperty {get;}
    public int SomeOtherPropery {get;}

    public SomeCommand(string someProperty, int someOtherProperty)
    {
        SomeProperty = someProperty;
        SomeOtherProperty = someOtherProperty;
    }
}
```
Instead of this:
```
public class SomeCommand
{
    public string SomeProperty {get; set;}
    public int SomeOtherPropery {get; set;}

    public SomeCommand()
    {
    }
}
```
And be able to use that POCO in an input binding.

As I said, I don't know if anything has changed in this regard since I created this issue 10 months ago, but back then, this was not possible because the Azure Functions runtime required parameterless constructor and public setters on all properties.

**Comment by fabiocav** [comment]
@CasperWSchmidt this is still the case. The activation/deserialization still require a default constructor and public setters for the POCO.

**Comment by v-anvari** [comment]
Closing the issue as it an un-supported feature.

**Comment by CasperWSchmidt** [comment]
@v-anvari why close this as un-supported feature when it is a feature request?

IMHO it should be possible. Such objects should be immutable but this is not possible with the current requirement of default constructor and public setters.

**Comment by v-anvari** [comment]
Hi @CasperWSchmidt , Adding this to triaged and feature label to be considered as a feature request for later releases.

**Comment by CasperWSchmidt** [comment]
@v-anvari perhaps you should also re-open it?

**Comment by CasperWSchmidt** [comment]
Perhaps using .Net Core 5.0 will help this feature being implemented: https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-5-0#deserialize-to-immutable-classes-and-structs

**Comment by CasperWSchmidt** [comment]
Any news on this?

**Comment by CasperWSchmidt** [comment]
I just tested it on V4 running .Net 6.0 and still no luck :( This is absurd! How can Microsoft expect System.Text.Json to be the default anywhere before it reaches at least feature parity with Newtonsoft.Json on such simple requirements? According to [this](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-immutability?pivots=dotnet-6-0) it should work, so what going on with Azure Functions/Webjobs?
